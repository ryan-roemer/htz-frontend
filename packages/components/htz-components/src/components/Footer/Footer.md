<!-- START doctoc generated TOC please keep comment here to allow auto update -->

<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

**Table of Contents**

* [Features](#features)
  * [**Basic use**](#basic-use)
  * [**DOM Element Attributes**](#dom-element-attributes)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

`<Footer>` is a responsive, flexbox-based, intended to be both
flexible and easy to use. No custom user props needed to be passed through.

### Features

* Responsive

#### **Basic use**

```jsx
<Footer />
```
