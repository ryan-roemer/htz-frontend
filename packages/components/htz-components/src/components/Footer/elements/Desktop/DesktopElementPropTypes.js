import PropTypes from 'prop-types';

export const PairTypes = PropTypes.arrayOf(
  PropTypes.shape({
    text: PropTypes.string,
    href: PropTypes.string,
  })
);

export const ColumnTypes = PropTypes.arrayOf(
  PropTypes.shape({
    /** Indicate list combination, if true the next list will be attached in the same column */
    combineWithNextColumn: PropTypes.bool,
    /** Inner Lists items passed from polopoly */
    items: PairTypes,
    // todo: ask if Pair type is needed (text and link options should pass?)
    /** List title  */
    title: PropTypes.string,
  })
);
