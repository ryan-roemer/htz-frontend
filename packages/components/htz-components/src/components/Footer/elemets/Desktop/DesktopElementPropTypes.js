import PropTypes from 'prop-types';

export const PairTypes = PropTypes.arrayOf(
  PropTypes.shape({
    text: PropTypes.string,
    href: PropTypes.string,
  })
);

export const ColumnTypes = PropTypes.arrayOf(
  PropTypes.shape({
    combineWithNextColumn: PropTypes.bool,
    items: PairTypes,
    title: PropTypes.string,
  })
);
