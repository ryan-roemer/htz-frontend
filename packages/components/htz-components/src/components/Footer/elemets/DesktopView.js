import React, { Fragment, } from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import { borderBottom, } from '@haaretz/htz-css-tools';
import { graphql, } from 'react-apollo';
import gql from 'graphql-tag';
import ButtonFooter from '../../Button/Button';
import Link from '../../Link/Link';
import IconFaceBookLogo from '../../Icon/icons/IconFacebookLogo';
import IconTwitter from '../../Icon/icons/IconTwitter';
import IconGPlus from '../../Icon/icons/IconGPlus';
import IconRss from '../../Icon/icons/IconRss';
import IconMailFooter from '../../Icon/icons/IconMailFooter';
import IconApple from '../../Icon/icons/IconApple';
import IconAndroid from '../../Icon/icons/IconAndroid';

const headLinkStyle = ({ theme, isLast, }) => ({
  ':after': {
    content: isLast ? '""' : '" | "',
    marginRight: '0.5rem',
  },
  // todo: ask for accurate margin left and right
  marginLeft: '0.5rem',
  fontWeight: 'bold',
});

const StyledHeadLink = createComponent(headLinkStyle, Link, [ 'content', 'href', ]);

const listLinkStyle = ({ theme, isLast, isBold = false, }) => ({
  ...(isBold ? { fontWeight: 'bold', } : {}),
  extend: [ theme.type(-1), ],
});

const StyledListLink = createComponent(listLinkStyle, Link, [ 'content', 'href', 'ref', ]);

const headWrapperLinkStyle = ({ theme, }) => ({
  display: 'flex',
  flexWrap: 'wrap',
});

const StyledHeadLinksWrapper = createComponent(headWrapperLinkStyle);

// todo: instead of wrapper use LayoutRow and LayoutContainer
const wrapperStyle = ({ theme, }) => ({
  paddingTop: '3rem',
  paddingBottom: '3rem',
  // todo: remove left and right padding, add a LayoutContainer component that will control max width
  paddingRight: '4rem',
  paddingLeft: '4rem',
  backgroundColor: theme.color('footer', 'bg'),
  color: theme.color('footer', 'text'),
});
const Wrapper = createComponent(wrapperStyle);

const desktopHeadStyle = ({ theme, }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'baseline',
  ...borderBottom('1px', '2', 'solid', 'white'),
});
const StyledDesktopHead = createComponent(desktopHeadStyle);

const desktopMainListStyle = ({ theme, }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  marginBottom: '4rem',
  marginTop: '2rem',
});
const StyledDesktopMainList = createComponent(desktopMainListStyle);

const desktopTextStyle = ({ theme, }) => ({
  extend: [ theme.type(-2), ],
});

// todo: change from div to p or something else?
const StyledDesktopText = createComponent(desktopTextStyle);

const expandedListContStyle = ({ theme, }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  paddingBottom: '6rem',
  ...borderBottom('1px', '2', 'solid', 'white'),
  marginBottom: '2rem',
});

const StyledExpandedListsCont = createComponent(expandedListContStyle);

const expandedListStyle = ({ theme, }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  flexWrap: 'wrap',
  alignItems: 'baseline',
});

const StyledExpandedLists = createComponent(expandedListStyle);

const listUlStyle = ({ theme, }) => ({
  marginInlineEnd: '12rem',
  paddingTop: '3rem',
  paddingBottom: '3rem',
});

const StyledLinkUl = createComponent(listUlStyle, 'ul');

const toolboxListStyle = () => ({
  minWidth: '20rem',
});

const StyledToolboxList = createComponent(toolboxListStyle, StyledLinkUl);

const titleLiStyle = ({ theme, }) => ({
  fontWeight: 'bold',
});

const StyledTitleLi = createComponent(titleLiStyle, 'li');

const IconMiscStyle = {
  marginRight: '3.5rem',
};

const PairTypes = PropTypes.arrayOf(
  PropTypes.shape({
    text: PropTypes.string,
    href: PropTypes.string,
  })
);

const ColumnTypes = PropTypes.arrayOf(
  PropTypes.shape({
    combineWithNextColumn: PropTypes.bool,
    items: PairTypes,
    title: PropTypes.string,
  })
);

class DesktopView extends React.Component {
  static propTypes = {
    Footer: PropTypes.shape({
      /** Indicates data loading state */
      loading: PropTypes.bool,
      /** Indicates data error state */
      error: PropTypes.bool,
      /** Footer data */
      footer: PropTypes.shape({
        head: PairTypes,
        columns: ColumnTypes,
        credit: PairTypes,
        toolbox: PairTypes,
      }),
    }).isRequired,
  };
  state = {
    expanded: false,
  };
  // shouldComponentUpdate(nextProps, nextState) {
  //   return this.props !== nextProps || this.state.expanded !== nextState.expanded;
  // }

  handleClick = () => {
    this.setState(prevState => ({
      expanded: !prevState.expanded,
    }));
    this.firstLiEl.focus();
  };
  render() {
    const { footer, loading, } = this.props.Footer;
    const { expanded, } = this.state;
    if (loading) {
      return <div> Loading... </div>;
    }
    const columnsArr = footer.columns.reduce((r, e, i, arr) => {
      const prev = arr[i - 1];
      if (prev && prev.combineWithNextColumn) r[r.length - 1].push(e);
      else r.push([ e, ]);
      return r;
    }, []);

    return (
      <Wrapper>
        <StyledDesktopHead>
          <div>הארץ</div>
          <div>
            <IconFaceBookLogo size={3} miscStyles={IconMiscStyle} />
            <IconTwitter size={3} miscStyles={IconMiscStyle} />
            <IconAndroid size={3} miscStyles={IconMiscStyle} />
            <IconApple size={3} miscStyles={IconMiscStyle} />
            <IconGPlus size={3} miscStyles={IconMiscStyle} />
            <IconMailFooter size={3} miscStyles={IconMiscStyle} />
            <IconRss size={3} miscStyles={IconMiscStyle} />
          </div>
        </StyledDesktopHead>
        <StyledDesktopMainList>
          <StyledHeadLinksWrapper>
            {footer.head.map((item, index) => (
              <StyledHeadLink
                content={item.text}
                href={item.href}
                isLast={index === footer.head.length - 1}
              />
            ))}
          </StyledHeadLinksWrapper>
          <div>
            <ButtonFooter
              variant="secondary"
              boxModel={{ hp: 4.5, vp: 0.75, }}
              onClick={() => this.handleClick()}
              attrs={{
                'aria-exapnded': expanded ? 'true' : 'false',
              }}
            >
              {expanded ? 'סגור' : 'הצג עוד'}
            </ButtonFooter>
          </div>
        </StyledDesktopMainList>
        {expanded ? (
          <StyledExpandedListsCont>
            <StyledExpandedLists>
              {columnsArr.map((lists, colIdx) => (
                <StyledLinkUl>
                  {lists.map((innerList, listIdx) => (
                    <div>
                      <StyledTitleLi>{innerList.title}</StyledTitleLi>
                      {innerList.items.map(link => (
                        <li>
                          <StyledListLink
                            content={link.text}
                            href={link.href}
                            {...(colIdx === 0 && listIdx === 0
                              ? {
                                ref: firstLiEl => {
                                  this.firstLiEl = firstLiEl;
                                },
                              }
                              : {})}
                          />
                        </li>
                      ))}
                    </div>
                  ))}
                </StyledLinkUl>
              ))}
            </StyledExpandedLists>
            <StyledToolboxList>
              {footer.toolbox.map(link => (
                <li>
                  <StyledListLink content={link.text} href={link.href} isBold />
                </li>
              ))}
            </StyledToolboxList>
          </StyledExpandedListsCont>
        ) : null}
        <StyledDesktopText>
          חדשות, ידיעות מהארץ והעולם - הידיעות והחדשות בעיתון הארץ. סקופים, מאמרים, פרשנויות ותחקירי
          עומק באתר האיכותי בישראל
        </StyledDesktopText>
        <StyledDesktopText>© כל הזכויות שמורות להוצאת עיתון הארץ בעמ</StyledDesktopText>
      </Wrapper>
    );
  }
}

export default graphql(
  gql`
    {
      footer {
        head {
          text
          href
        }
        columns {
          title
          combineWithNextColumn
          items {
            text
            href
          }
        }
        credit {
          text
          href
        }
        toolbox {
          text
          href
        }
      }
    }
  `,
  { name: 'Footer', }
)(DesktopView);
