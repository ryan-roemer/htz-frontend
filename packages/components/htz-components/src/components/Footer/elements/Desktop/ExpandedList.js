import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import { borderBottom, } from '@haaretz/htz-css-tools';
import Link from '../../../Link/Link';
import Grid from '../../../Grid/Grid';
import GridItem from '../../../Grid/GridItem';
import { ColumnTypes, PairTypes, } from './DesktopElementPropTypes';

const extendedListContainerStyle = ({
  theme: { color, footerBorderStyle: { borderWidth, lines, borderStyle, }, },
  showMe,
}) => ({
  display: showMe ? 'flex' : 'none',
  backgroundColor: color('footer', 'bg'),
  alignItems: 'baseline',
  paddingBottom: '3rem',
  marginBottom: '2rem',
  ...borderBottom(borderWidth, 0, borderStyle, color('footer', 'border')),
});

const ExtendedFooterContainer = createComponent(extendedListContainerStyle);

const listUlStyle = ({ theme, }) => ({
  ...theme.mq({ from: 'xl', }, { marginInlineEnd: '8rem', }),
  marginBottom: '2rem',
});

const StyledSection = createComponent(listUlStyle, 'section');

const titleLiStyle = ({ theme, }) => ({
  fontWeight: 'bold',
});

const StyledUlTitle = createComponent(titleLiStyle, 'h4');

const listLinkStyle = ({ theme, isLast, isBold = false, }) => ({
  ...(isBold ? { fontWeight: 'bold', } : {}),
  extend: [ theme.type(-1), ],
});

const StyledListLink = createComponent(listLinkStyle, Link, [
  'content',
  'href',
  'focus',
]);

ExpandedList.propTypes = {
  /** Footer's array of columns data */
  columnsArr: PropTypes.arrayOf(ColumnTypes).isRequired,
  /** Indicates display mode */
  showMe: PropTypes.bool.isRequired,
  /** Footer's toolbox data */
  toolbox: PairTypes.isRequired,
};

function ExpandedList({ columnsArr, toolbox, showMe, }) {
  return (
    <ExtendedFooterContainer showMe={showMe}>
      <Grid>
        <GridItem
          width={[
            { from: 'm', until: 'l', value: 2 / 3, },
            { from: 'l', until: 'xl', value: 4 / 5, },
            { from: 'xl', value: 5 / 6, },
          ]}
        >
          <Grid>
            {columnsArr.map((lists, colIdx) => (
              <GridItem
                width={[
                  { from: 'm', until: 'l', value: 1 / 2, },
                  { from: 'l', until: 'xl', value: 1 / 4, },
                  { from: 'xl', value: 1 / 5, },
                ]}
                // eslint-disable-next-line  react/no-array-index-key
                key={colIdx}
              >
                {lists.map((innerList, listIdx) => (
                  // eslint-disable-next-line react/no-array-index-key
                  <StyledSection key={listIdx}>
                    <StyledUlTitle>{innerList.contentName}</StyledUlTitle>
                    <ul>
                      {innerList.row.map((link, linkIdx) => (
                        <li key={`${link.contentName}${link.value}`}>
                          <StyledListLink
                            content={link.contentName}
                            href={link.value}
                          />
                        </li>
                      ))}
                    </ul>
                  </StyledSection>
                ))}
              </GridItem>
            ))}
          </Grid>
        </GridItem>
        <GridItem
          width={[
            { from: 'm', until: 'l', value: 1 / 3, },
            { from: 'l', until: 'xl', value: 1 / 5, },
            { from: 'xl', value: 1 / 6, },
          ]}
        >
          <ul>
            {toolbox.map(link => (
              <li key={`${link.text}${link.value}`}>
                <StyledListLink
                  content={link.contentName}
                  href={link.value}
                  isBold
                />
              </li>
            ))}
          </ul>
        </GridItem>
      </Grid>
    </ExtendedFooterContainer>
  );
}

export default ExpandedList;
