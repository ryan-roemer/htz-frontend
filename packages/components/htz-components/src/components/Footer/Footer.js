import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, withTheme, } from 'react-fela';
import { graphql, } from 'react-apollo';
import gql from 'graphql-tag';
import ButtonFooter from '../Button/Button';
import Link from '../Link/Link';
import LayoutFooterRow from '../PageLayout/LayoutRow';
import LayoutFooterContainer from '../PageLayout/LayoutContainer';
import ExpandedList from './elements/Desktop/ExpandedList';
import {
  ColumnTypes,
  PairTypes,
} from './elements/Desktop/DesktopElementPropTypes';
// Views ///////////////////////////////////////////////////////////////////
import FooterHead from './elements/FooterHead';
import MobileView from './elements/MobileMainContainer';
// ///////////////////////////////////////////////////////////////////

const desktopMainListLayoutContainerStyle = {
  display: 'flex',
  flexDirection: 'column',
  paddingTop: [ { from: 's', value: '13rem', }, ],
  paddingBottom: [ { from: 's', value: '8rem', }, { until: 's', value: '4rem', }, ],
};

const headLinkStyle = ({ theme, isLast, }) => ({
  marginLeft: '1rem',
  fontWeight: 'bold',
  ':after': {
    content: isLast ? '""' : '"|"',
    marginRight: '1rem',
  },
});

const StyledHeadLink = createComponent(headLinkStyle, Link, [
  'content',
  'href',
]);

const ListUlStyle = () => ({
  marginInlineEnd: '2rem',
});
const StyledUlLinks = createComponent(ListUlStyle, 'ul');

const ListLiStyle = () => ({
  display: 'inline-block',
});
const StyledLi = createComponent(ListLiStyle, 'li');

const desktopBodyStyle = ({ theme, }) => ({
  extend: [
    {
      ...theme.mq(
        {
          until: 's',
        },
        {
          display: 'none',
        }
      ),
    },
  ],
});
const StyledDesktopBody = createComponent(desktopBodyStyle);

const headWrapperLinkStyle = ({ theme, }) => ({
  display: 'flex',
  flexWrap: 'no-wrap',
  alignItems: 'center',
  marginTop: '2rem',
  marginBottom: '4rem',
  extend: [ theme.mq({ until: 'l', }, { alignItems: 'start', }), ],
});

const StyledHeadLinksWrapper = createComponent(headWrapperLinkStyle);

const optionalExtendedWrapper = ({ theme, }) => ({
  extend: [ theme.type(-2), ],
});
const StyledDesktopText = createComponent(optionalExtendedWrapper);

export class Footer extends React.Component {
  static propTypes = {
    Footer: PropTypes.shape({
      /** Indicates data loading state */
      loading: PropTypes.bool,
      /** Indicates data error state */
      error: PropTypes.bool,
      /** Footer data from polopoly */
      footer: PropTypes.shape({
        head: PairTypes,
        columns: ColumnTypes,
        credit: PairTypes,
        toolbox: PairTypes,
      }),
    }).isRequired,
    theme: PropTypes.shape({
      footerDesktopI18n: PropTypes.shape({
        ExpandedButton: PropTypes.object,
        Copyright: PropTypes.object,
      }),
      color: PropTypes.func.isRequired,
    }).isRequired,
  };
  state = {
    expanded: false,
  };

  handleClick = () => {
    this.setState(prevState => ({
      expanded: !prevState.expanded,
    }));
    this.focusElement.focus();
  };

  render() {
    const { theme, } = this.props;
    const {
      theme: { footerDesktopI18n: { ExpandedButton, Copyright, }, color, },
      Footer: { footer, loading, },
    } = this.props;

    const { expanded, } = this.state;
    if (loading) {
      return <div> Loading... </div>;
    }
    const columnsArr = footer.columns.reduce((r, e, i, arr) => {
      const prev = arr[i - 1];
      if (prev && prev.combineWithNextColumn) r[r.length - 1].push(e);
      else r.push([ e, ]);
      return r;
    }, []);
    console.warn(columnsArr);
    return (
      <LayoutFooterRow
        bgc={color('footer', 'bg')}
        miscStyles={{
          color: color('footer', 'text'),
          paddingInlineStart: '8rem',
          paddingInlineEnd: '8rem',
        }}
        tagName="footer"
      >
        <LayoutFooterContainer
          miscStyles={desktopMainListLayoutContainerStyle}
          bgc={color('footer', 'bg')}
        >
          <FooterHead />
          <MobileView theme={theme} />
          <StyledDesktopBody>
            <StyledHeadLinksWrapper>
              <StyledUlLinks>
                {footer.head.map((item, index) => (
                  <StyledLi key={`${item.contentName}${item.value}`}>
                    <StyledHeadLink
                      key={`${item.contentName}${item.value}`}
                      content={item.contentName}
                      href={item.value}
                      isLast={index === footer.head.length - 1}
                    />
                  </StyledLi>
                ))}
              </StyledUlLinks>
              <ButtonFooter
                variant="secondary"
                boxModel={{ hp: 4, vp: 1, }}
                onClick={() => this.handleClick()}
                attrs={{
                  'aria-expanded': expanded ? 'true' : 'false',
                }}
                miscStyles={{ marginInlineStart: 'auto', }}
              >
                {expanded ? ExpandedButton.close : ExpandedButton.showMore}
              </ButtonFooter>
            </StyledHeadLinksWrapper>
            <div
              ref={el => {
                this.focusElement = el;
              }}
              tabIndex="-1"
            >
              <ExpandedList
                toolbox={footer.toolbox}
                columnsArr={columnsArr}
                showMe={expanded}
              />
            </div>
            <StyledDesktopText>{Copyright.firstRow}</StyledDesktopText>
            <StyledDesktopText>{Copyright.secondRow}</StyledDesktopText>
          </StyledDesktopBody>
        </LayoutFooterContainer>
      </LayoutFooterRow>
    );
  }
}
const footerWithTheme = withTheme(Footer);

export default graphql(
  gql`
    {
      footer {
        head {
          contentName
          value
        }
        columns {
          contentName
          combineWithNextColumn
          row {
            contentName
            value
          }
        }
        credit {
          contentName
          value
        }
        toolbox {
          contentName
          value
        }
      }
    }
  `,
  { name: 'Footer', }
)(footerWithTheme);
