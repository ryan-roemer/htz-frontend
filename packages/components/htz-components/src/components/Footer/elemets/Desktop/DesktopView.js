import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import { borderBottom, } from '@haaretz/htz-css-tools';
import { graphql, } from 'react-apollo';
import gql from 'graphql-tag';
import ButtonFooter from '../../../Button/Button';
import Link from '../../../Link/Link';
import LayoutFooterRow from '../../../PageLayout/LayoutRow';
import LayoutFooterContainer from '../../../PageLayout/LayoutContainer';
import IconFaceBookLogo from '../../../Icon/icons/IconFacebookLogo';
import IconHaaretzLogo from '../../../Icon/icons/IconHaaretzLogo';
import IconTwitter from '../../../Icon/icons/IconTwitter';
import IconGPlus from '../../../Icon/icons/IconGPlus';
import IconRss from '../../../Icon/icons/IconRss';
import IconMailFooter from '../../../Icon/icons/IconMailFooter';
import IconApple from '../../../Icon/icons/IconApple';
import IconAndroid from '../../../Icon/icons/IconAndroid';
import ExpandedList from './ExpandedList';
import { ColumnTypes, PairTypes, } from './DesktopElementPropTypes';

const desktopMainListLayoutContainerStyle = {
  display: 'flex',
  flexDirection: 'column',
  paddingTop: '10rem',
  paddingBottom: '10rem',
};

const desktopHeadStyle = ({
  theme: { color, footerBorderStyle: { borderWidth, lines, borderStyle, }, },
}) => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'baseline',
  ...borderBottom(borderWidth, lines, borderStyle, color('footer', 'border')),
});
const StyledDesktopHead = createComponent(desktopHeadStyle);

const headLinkStyle = ({ theme, isLast, }) => ({
  ':after': {
    content: isLast ? '""' : '"|"',
    marginRight: '1rem',
  },
  marginLeft: '1rem',
  fontWeight: 'bold',
});

const StyledHeadLink = createComponent(headLinkStyle, Link, [ 'content', 'href', ]);

const ListLiStyle = () => ({
  display: 'inline-block',
});
const StyledLi = createComponent(ListLiStyle, 'li');

const ListUlStyle = () => ({
  marginInlineEnd: '2rem',
});
const StyledUlLinks = createComponent(ListUlStyle, 'ul');

const headWrapperLinkStyle = ({ theme, }) => ({
  display: 'flex',
  flexWrap: 'no-wrap',
  justifyContent: 'space-between',
  alignItems: 'center',
  marginTop: '1rem',
  marginBottom: '5rem',
});

const StyledHeadLinksWrapper = createComponent(headWrapperLinkStyle);

const optionalExtendedWrapper = ({ theme, }) => ({
  extend: [ theme.type(-2), ],

});
const StyledDesktopText = createComponent(optionalExtendedWrapper);

const IconMiscStyle = {
  marginInlineStart: '3.5rem',
};
// todo: ask if there is better fix for logo when svg shape is not square
const LogoMiscStyle = {
  marginBottom: '-5rem',
  marginTop: '-5.5rem',
  marginInlineStart: '-0.5rem',
};

export class DesktopView extends React.Component {
  static propTypes = {
    Footer: PropTypes.shape({
      /** Indicates data loading state */
      loading: PropTypes.bool,
      /** Indicates data error state */
      error: PropTypes.bool,
      /** Footer data */
      footer: PropTypes.shape({
        head: PairTypes,
        columns: ColumnTypes,
        credit: PairTypes,
        toolbox: PairTypes,
      }),
    }).isRequired,
    theme: PropTypes.shape({
      footerDesktopI18n: PropTypes.shape({
        ExpandedButton: PropTypes.object,
        Copyright: PropTypes.object,
      }),
      color: PropTypes.func.isRequired,
    }).isRequired,
  };
  state = {
    expanded: false,
  };

  handleClick = () => {
    this.setState(prevState => ({
      expanded: !prevState.expanded,
    }));
  }

  render() {
    const {
      theme: {
        footerDesktopI18n: { ExpandedButton, Copyright, },
        color,
      },
      Footer: { footer, loading, },
    } = this.props;
    const { expanded, } = this.state;
    if (loading) {
      return <div> Loading... </div>;
    }   
    const columnsArr = footer.columns.reduce((r, e, i, arr) => {
      const prev = arr[i - 1];
      if (prev && prev.combineWithNextColumn) r[r.length - 1].push(e);
      else r.push([ e, ]);
      return r;
    }, []);

    return (
      <LayoutFooterRow
        bgc={color('footer', 'bg')}
        miscStyles={{ color: color('footer', 'text'), paddingInlineStart: '8rem', paddingInlineEnd: '8rem', }}
      >
        <LayoutFooterContainer
          miscStyles={desktopMainListLayoutContainerStyle}
          bgc={color('footer', 'bg')}
        >
          <StyledDesktopHead>
            <div><IconHaaretzLogo size={17.7} miscStyles={LogoMiscStyle} /></div>
            <div>
              <Link content={<IconFaceBookLogo size={3} miscStyles={IconMiscStyle} />} href="https://www.facebook.com/haaretz" />
              <Link content={<IconTwitter size={3} miscStyles={IconMiscStyle} />} href="https://twitter.com/haaretz" />
              <Link content={<IconAndroid size={3} miscStyles={IconMiscStyle} />} href="https://play.google.com/store/apps/details?id=com.haaretz" />
              <Link content={<IconApple size={3} miscStyles={IconMiscStyle} />} href="https://itunes.apple.com/us/app/id521559643" />
              <Link content={<IconGPlus size={3} miscStyles={IconMiscStyle} />} href="https://plus.google.com/+haaretzcoil" />
              <Link content={<IconMailFooter size={3} miscStyles={IconMiscStyle} />} href="https://www.haaretz.co.il/misc/redemail" />
              <Link content={<IconRss size={3} miscStyles={IconMiscStyle} />} href="https://www.haaretz.co.il/misc/rss" />
            </div>
          </StyledDesktopHead>

          <StyledHeadLinksWrapper>
            <StyledUlLinks>
              {footer.head.map((item, index) => (
                <StyledLi key={`${item.text}${item.href}`}>
                  <StyledHeadLink
                    key={`${item.text}${item.href}`}
                    content={item.text}
                    href={item.href}
                    isLast={index === footer.head.length - 1}
                  />
                </StyledLi>
              ))}
            </StyledUlLinks>
            <StyledLi>
              <ButtonFooter
                variant="secondary"
                boxModel={{ hp: 4.5, vp: 0.75, }}
                onClick={() => this.handleClick()}
                attrs={{
                  'aria-expanded': expanded ? 'true' : 'false',
                }}
              >
                {expanded ? ExpandedButton.close : ExpandedButton.showMore}
              </ButtonFooter>
            </StyledLi>
          </StyledHeadLinksWrapper>

          <ExpandedList
            toolbox={footer.toolbox}
            columnsArr={columnsArr}
            showMe={expanded}
          />
          <StyledDesktopText>{Copyright.firstRow}</StyledDesktopText>
          <StyledDesktopText>{Copyright.secondRow}</StyledDesktopText>
        </LayoutFooterContainer>
      </LayoutFooterRow>
    );
  }
}

export default graphql(
  gql`
    {
      footer {
        head {
          text
          href
        }
        columns {
          title
          combineWithNextColumn
          items {
            text
            href
          }
        }
        credit {
          text
          href
        }
        toolbox {
          text
          href
        }
      }
    }
  `,
  { name: 'Footer', }
)(DesktopView);
