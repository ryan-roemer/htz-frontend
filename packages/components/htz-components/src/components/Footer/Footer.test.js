/* global window */
import React from 'react';
import matchMediaPolyfill from 'mq-polyfill';
import { withTheme, } from 'react-fela';
import felaSnapshotter from '../../test-helpers/felaSnapshotter';
// import { felaMount, felaShallow, } from '../../test-helpers/felaEnzymeRenderers';
import Footer from './Footer';
import { DesktopView, } from './elemets/Desktop/DesktopView';
// import MobileView from './elemets/MobileView';

const FooterDesktopMock = {
  /** Indicates data loading state */
  loading: false,
  /** Indicates data error state */
  error: false,
  /** Footer data */
  footer: {
    head: [ { text: 'מערכת', href: 'https://www.haaretz.co.il/misc/editors', }, ],
    columns: [
      {
        title: 'מדורים',
        combineWithNextColumn: true,
        items: [
          {
            text: 'חדשות',
            href: 'https://www.haaretz.co.il/news',
          },
        ],
      },
      {
        title: 'Haaretz.com',
        combineWithNextColumn: false,
        items: [
          {
            text: 'Israel - Syria rebels',
            href:
              'https://www.haaretz.com/israel-news/with-eye-on-iran-israel-increases-military-support-for-syrian-rebels-1.5826348',
          },
          {
            text: 'Freemasons',
            href:
              'https://www.haaretz.com/israel-news/MAGAZINE-not-a-cult-the-freemasons-want-you-unless-you-you-1.5824127',
          },
        ],
      },
    ],
    credit: [ { text: 'מערכת', href: 'https://www.haaretz.co.il/misc/editors', }, ],
    toolbox: [ { text: 'מערכת', href: 'https://www.haaretz.co.il/misc/editors', }, ],
  },
};
const DesktopFooterWithTheme = withTheme(DesktopView);

describe('<Footer /> ', () => {
  describe('Desktop view DOM elements ', () => {
    it('Render desktop Footer element correctly', () => {
      const { component, styles, } = felaSnapshotter(<DesktopFooterWithTheme Footer={FooterDesktopMock} />);
      expect(component).toMatchSnapshot();
      expect(styles).toMatchSnapshot();
    });
  });
  describe('Mobile view DOM elements ', () => {
    it('Render mobile Footer element correctly', () => {
      matchMediaPolyfill(window);
      window.resizeTo = resizeTo;
      window.resizeTo(400, 768);
      const { component, styles, } = felaSnapshotter(<Footer />);
      expect(component).toMatchSnapshot();
      expect(styles).toMatchSnapshot();
    });
  });
});

function resizeTo(width, height) {
  Object.assign(this, {
    innerWidth: width,
    innerHeight: height,
    outerWidth: width,
    outerHeight: height,
  }).dispatchEvent(new this.Event('resize'));
}
