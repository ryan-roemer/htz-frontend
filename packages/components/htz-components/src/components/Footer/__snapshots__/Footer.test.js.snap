// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer />  Desktop view DOM elements  Render desktop Footer element correctly 1`] = `
<div>
  <div
    className="FI_PH"
  />
  <div
    className="FI_PH"
  />
</div>
`;

exports[`<Footer />  Desktop view DOM elements  Render desktop Footer element correctly 2`] = `""`;

exports[`<Footer />  Mobile view DOM elements  Render mobile Footer element correctly 1`] = `
<div>
  <div
    className="FI_PH"
  />
  <div
    className="FI_PH"
  />
</div>
`;

exports[`<Footer />  Mobile view DOM elements  Render mobile Footer element correctly 2`] = `""`;
